packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit_webview_widget.dart

```diff
@@ -161,6 +164,112 @@ class WebKitWebViewPlatformController extends WebViewPlatformController {
     };
   }
 
+  @override
+  Future<void> loadHtmlString(String html, {String? baseUrl}) {
+    return webView.loadHtmlString(html, baseUrl);
+  }
+
+  @override
+  Future<void> loadFile(String absoluteFilePath) async {
+    await webView.loadFileUrl(
+      absoluteFilePath,
+      path.dirname(absoluteFilePath),
+    );
+  }
+
+  @override
+  Future<void> loadFlutterAsset(String key) async {
+    assert(key.isNotEmpty);
+    return webView.loadFlutterAsset(key);
+  }
+
+  @override
+  Future<void> loadUrl(String url, Map<String, String>? headers) async {
+    final NSUrlRequest request = NSUrlRequest(
+      url: url,
+      allHttpHeaderFields: headers ?? <String, String>{},
+    );
+    return webView.loadRequest(request);
+  }
+
+  @override
+  Future<void> loadRequest(WebViewRequest request) async {
+    if (!request.uri.hasScheme) {
+      throw ArgumentError('WebViewRequest#uri is required to have a scheme.');
+    }
+
+    final NSUrlRequest urlRequest = NSUrlRequest(
+      url: request.uri.toString(),
+      allHttpHeaderFields: request.headers,
+      httpMethod: describeEnum(request.method),
+      httpBody: request.body,
+    );
+
+    return webView.loadRequest(urlRequest);
+  }
+
+  @override
+  Future<bool> canGoBack() => webView.canGoBack;
+
+  @override
+  Future<bool> canGoForward() => webView.canGoForward;
+
+  @override
+  Future<void> goBack() => webView.goBack();
+
+  @override
+  Future<void> goForward() => webView.goForward();
+
+  @override
+  Future<void> reload() => webView.reload();
+
+  @override
+  Future<String> evaluateJavascript(String javascript) async {
+    return runJavascriptReturningResult(javascript);
+  }
+
+  @override
+  Future<void> runJavascript(String javascript) async {
+    await webView.evaluateJavaScript(javascript);
+  }
+
+  @override
+  Future<String> runJavascriptReturningResult(String javascript) async {
+    return await webView.evaluateJavaScript(javascript) ?? '';
+  }
+
+  @override
+  Future<String?> getTitle() => webView.title;
+
+  @override
+  Future<void> scrollTo(int x, int y) async {
+    webView.scrollView.contentOffset = Point<double>(
+      x.toDouble(),
+      y.toDouble(),
+    );
+  }
+
+  @override
+  Future<void> scrollBy(int x, int y) async {
+    final Point<double> offset = await webView.scrollView.contentOffset;
+    webView.scrollView.contentOffset = Point<double>(
+      offset.x + x.toDouble(),
+      offset.y + y.toDouble(),
+    );
```

Comment:
We should consider flexing the boundary here and putting scroll-by functionality on the webview; this method will do weird things if there's other scrolling happening at the same time due to the async gap between the getter and setter.

E.g., imagine that someone calls (scrollBy(0, 10), and the JS on the page does its own scroll-by 50y which happens to fire at ~the same time. With the current (native) implementation, the result is that the page scrolls 60. With this implementation, the result will either be that it scrolls by 60, or that it scrolls by 10 (possibly with a visible flicker to 50), depending on where the JS fires relative to the async calls.

Reply:
I think this is caused by the fact that `scrollBy` is asynchronous and not that there are multiple calls. Even after I added a `scrollBy` method to `UIScrollView`, an app user will still be able to scroll while the `scrollBy` method call is being passed over. Leading to the same problem, right? I would also suspect that `scrollBy` is often used after calling `getScrollX` and `getScrollY` anyways. 

I think async message channels just makes this method inherently flawed compared to just using `scrollTo`.

Reply:
That also makes me wonder if we should change `getScrollX` and `getScrollY` to `getScrollPosition` in the new interface.

Reply:
> Even after I added a `scrollBy` method to `UIScrollView`, an app user will still be able to scroll while the `scrollBy` method call is being passed over. Leading to the same problem, right?

If it's a single call over the method channel, no scrolling can be lost even when it's async. Regardless of what the user is doing around when it arrives, it will still scroll by the requested amount.

In my example scenario, here are the possible outcomes with the multi-async-call implementation:

```
other scroll happens
* plugin gets position
* plugin sets position
=> Total scroll is the sum of both scrolls

* plugins gets position
other scroll happens
* plugin sets position
=> Total scroll is 10; the other scroll is destroyed

* plugin gets position
* plugin sets position
other scroll happens
= Total scroll is the sum of both scrolls
```

(If there is a stream of other scroll events, the chances of some of them being in the middle and lost go up significantly.)

With the single async method, the other scroll can only be entirely before or entirely after, so the lost scroll can't happen.

> I would also suspect that scrollBy is often used after calling getScrollX and getScrollY anyways.

Why wouldn't people use scrollTo for that use case? I would expect scrollBy to be for something like a "page down" button/key, where the delta is all that matters.

But yes, if people are trying to use getScroll + scrollBy as a way of getting scrollTo behavior, they may get unexpected results due to async. That's already true though.

Reply:
> That also makes me wonder if we should change `getScrollX` and `getScrollY` to `getScrollPosition` in the new interface.

I think that makes a lot of sense.